README.md
# wiki_summary_extract
Personal Project Python module 

Creator: Josue Lopez

Installation: 
  - Download Python 3+
 
Usage:
  - Open Terminal
  - Change directory into github project
  - run `python3 wiki_summary.py`
  - Follow the programs instruction to get your search result


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### 1.2.8 (2020-10-23)

### 1.2.7 (2020-10-23)

### 1.2.6 (2020-10-23)

### 1.2.5 (2020-09-10)


### Bug Fixes

* Setup action python 3.x ([82e0d8d](https://github.com/josueal1/wiki_summary_extract/commit/82e0d8df6e8e088e4c2315f4cb93cbb93d1c87ce))

### 1.2.4 (2020-09-10)

### 1.2.3 (2020-09-10)


### Bug Fixes

* test printing pip version ([e452b92](https://github.com/josueal1/wiki_summary_extract/commit/e452b92b6c7bf8cceba11ec1fd8599664e80ffb2))

### 1.2.2 (2020-08-12)

### 1.2.1 (2020-08-12)

## [1.2.0](https://github.com/josueal1/wiki_summary_extract/compare/v1.1.0...v1.2.0) (2020-08-12)


### Features

* **app:** Mention underscores in prompt ([2b0bde6](https://github.com/josueal1/wiki_summary_extract/commit/2b0bde680d214de0d2746595f947062e0ed1981a))

## 1.1.0 (2020-08-12)


### Features

* Add creator name to README.md ([d6dff87](https://github.com/josueal1/wiki_summary_extract/commit/d6dff87d3cb741d19bb3134f2cdb3ddd5c60a516))
* **README.md:** Add installation and usage steps ([15a6a3a](https://github.com/josueal1/wiki_summary_extract/commit/15a6a3a6f1c2a0cc5b3da091d3eaeb8896c37206))


### Bug Fixes

* **.gitignore:** Ignore .DS_Store file ([c5dd940](https://github.com/josueal1/wiki_summary_extract/commit/c5dd94062d63a0903cd8d7a002068051acbe4388))

## 1.0.0 (2020-08-12)


### Features

* Add creator name to README.md ([d6dff87](https://github.com/josueal1/wiki_summary_extract/commit/d6dff87d3cb741d19bb3134f2cdb3ddd5c60a516))


src/scripts/lint.sh
#!/bin/bash
# Linter are static code analysis tools that flags syntax errors, bugs, style

printf "\nlint.sh: Starting...\n"

# if this script called and given any commandline args 
if [[ $# > 2 ]];then
	# Extract array arg passed down via commandline
	ARRAY=( "$@" )
	last_idx=$(( ${#ARRAY[@]} - 1 ))
	current_branch=${ARRAY[$last_idx]}
	unset ARRAY[$last_idx]
else
	current_branch=$(git branch | sed -n -e 's/^\* \(.*\)/\1/p');

	# get list of files that were 'git add' or 'git commit' and don't exist on master
	ARRAY=($(git diff origin/master $current_branch --name-only HEAD));
fi

printf "\nFollowing files are differnet between 'origin/master' and '${current_branch}':\n\n"
printf "%s\n" "${ARRAY[@]}"


# Get the number of Python files currently Git Added
NUM_PY_FILES=0
for i in "${ARRAY[@]}"
do
	if [[ "$i" == *".py"* ]]; then
		((NUM_PY_FILES++))
	fi
done

printf "\n"
printf "lint.sh: $NUM_PY_FILES python files git add'ed.\n\n"

# L
for filename in "${ARRAY[@]}"
do
	ERROR_CODES="E1101"
	if [[ "$filename" == *".py"* ]]; then

		pylint "$filename" --exit-zero --jobs=0 --score=y \
			--msg-template="{msg_id}: {category}: {path} {line}:{column} {msg}" \
			--disable=$ERROR_CODES		
	fi
done

printf "lint.sh: Done\n\n"


wiki_summary_extract.py
# Creator Josue Lopez
# Mon Jul 16 10:30 PM
# Python 3.6

import urllib.request
import urllib.parse
import urllib.error
import json

"""
This module is a basic script that searches for a simple summary
of anything using Wikimedia's REST API. It's a basic interface
search tool written in Python 3.6.

When the program is executed, it will run the 'main' function
which handldes the entire module.
"""

"""
The use of Wikimedia REST API is solely for personal / education purposes, and
I agree to Wikimedia's general Term's of Use and Private policy.
"""

def main():
    title = input("Search for a famous name on wikipedia: ")
    status = None
    try:
        print('...')
        url = urllib.request.urlopen("https://en.wikipedia.org/api/rest_v1/page/summary/"+title)
        status = True
    except urllib.error.HTTPError:
        print("\nSorry, an error occurred. Our wiki_extract module believes this search does not exist in Wikimedia's database.\n")
    else:
        response = url.read().decode(encoding ='utf-8')

        json_data = json.loads(response)

        print('\n' + json_data['displaytitle'] + '\n')
        print(json_data['extract'].encode('ascii', 'ignore').decode('ascii') + '\n')
    finally:
        print("Success.") if status is True else print('Try Again.')
        print("Done.")

if __name__ == '__main__':
    main()


wiki_summary.py
"""
This module is a basic script that searches for a simple summary
of anything using Wikimedia's REST API. It's a basic interface
search tool written in Python 3.6.

When the program is executed, it will run the 'main' function
which handldes the entire module.

The use of Wikimedia REST API is solely for personal / education purposes, and
I agree to Wikimedia's general Term's of Use and Private policy.
"""

# Creator Josue Lopez
# Mon Jul 16 2018 11:50 PM
# Python 3.6

import urllib.request
import urllib.parse
import urllib.error
import json


def main():
    """ Main function that runs the CLI app """
    title = input("Search for a famous name on wikipedia with underscore as \
        spaces (e.j. Barrack_Obama): ")
    status = None
    try:
        print('...')
        wiki_summary_uri = "https://en.wikipedia.org/api/rest_v1/page/summary/"
        url = urllib.request.urlopen(wiki_summary_uri+title)
        status = True
    except urllib.error.HTTPError:
        print("\nSorry, an error occurred. Our wiki_extract module believes \
            this search does not exist in Wikimedia's database.\n")
    else:
        response = url.read().decode(encoding='utf-8')

        json_data = json.loads(response)

        print('\n' + json_data['displaytitle'] + '\n')
        data = json_data['extract'].encode('ascii', 'ignore').decode('ascii')
        print(data + '\n')
    finally:
        if status is True:
            print("Success.")
        else:
            print('Try Again.')

        print("Done.")

if __name__ == '__main__':
    main()
